Finish up the RTM for the framework

permcody opened the pull request:
refs #9554

<!--
INCLUDE THE FOLLOWING IN THE PR DESCRIPTION
- Explain relevant design information for your change.
- Follow the [Coding Standards](http://mooseframework.org/wiki/CodeStandards/).
- Submit or improve [Test Cases](http://mooseframework.org/wiki/MooseTraining/testing/).
- Reference a specific issue, place "refs #<issue>" or "closes #<issue>" (e.g., #closes #1234).
-->

permcody committed:
Finish up the RTM for the framework

refs #9554

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/444172/) on 391dea8 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/14522/site)

This comment will be updated on new commits.

permcody committed:
Move minimum_requirements.md doc

Moving the min requirements so that we can access from framework builds
as well as module builds.

moosebuild commented:
Job [Modules debug PETSc alt](https://civet.inl.gov/job/444384/) on 3e681d5 : invalidated by @permcody

Timeout

permcody commented on framework/doc/content/sqa/assumptions_and_dependencies.md:
```diff
@@ -0,0 +1,6 @@
+The software should be designed with the fewest possible constraints. Ideally the software should run
+on a wide variety of evolving hardware so it should follow well-adopted standards and guidelines. The
+software should run on any [!ac](POSIX) compliant system. The software will also make use FEM and
```
```suggestion
software should run on any [!ac](POSIX) compliant system. The software will also make use of FEM and
```
